File: Daily_Codes_Answers/day-10/13_Roman_to_Integer.py
def func(s):
    value=0
    
    I=1
    V=5
    X=10
    L=50
    C=100
    D=500
    M=1000

    if "IV" in s:
        value+=4
    if "IX" in s:
        value+=9
    if "XL" in s:
        value+=40
    if "XC" in s:
        value+=90
    if "CD" in s:
        value+=400
    if "CM" in s:
        value+=900
            
    value += (s.count("I")-s.count("IV")-s.count("IX"))*I
    value += (s.count("V")-s.count("IV"))*V
    value += (s.count("X")-s.count("IX")-s.count("XL")-s.count("XC"))*X
    value += (s.count("L")-s.count("XL"))*L
    value += (s.count("C")-s.count("CD")-s.count("CM")-s.count("XC"))*C
    value += (s.count("D")-s.count("CD"))*D
    value += (s.count("M")-s.count("CM"))*M
    
    return value

s1 = "III"
s2 = "LVIII"
s3 = "MCMXCIV"

print(f"test case 1 ---> {s1} = {func(s1)}")
print(f"test case 2 ---> {s2} = {func(s2)}")
print(f"test case 3 ---> {s3} = {func(s3)}")

File: Daily_Codes_Answers/day-10/13_Roman_to_Integer.js
const romanToInt = (s)=>{
    let I=1;
    let X=10;
    let C=100;

    let value=0;
    for(let i=s.length-1 ; i>=0 ; i--){
        if(s.charAt(i)=='I'){
            value+=I;
            continue;
        }else{
            I=-1;
        }
        if(s.charAt(i)=='V'){
            value+=5;
            continue;
        }
        if(s.charAt(i)=='X'){
            value+=X;
            continue;
        }else{
            X=-10;
        }
        if(s.charAt(i)=='L'){
            value+=50;
            continue;
        }
        if(s.charAt(i)=='C'){
            value+=C;
            continue;
        }else{
            C=-100;
        }
        if(s.charAt(i)=='D'){
            value+=500;
            continue;
        }
        if(s.charAt(i)=='M'){
            value+=1000;
            continue;
        }
    }
    return value;
}
let s1 = "III";
let s2 = "LVIII";
let s3 = "MCMXCIV";

console.log("test case 1 ---> "+s1+" = "+romanToInt(s1));
console.log("test case 2 ---> "+s2+" = "+romanToInt(s2));
console.log("test case 3 ---> "+s3+" = "+romanToInt(s3));